### PR TITLE
Remove `tox-pip-sync`

### PR DESCRIPTION
**Pull request by seanh**
Same as was done for `h-periodic`:

https://github.com/hypothesis/h-periodic/pull/232/

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,6 +70,43 @@ run-docker:
...
+#
+# This needs to be able to re-create requirements/*.txt files that don't exist
+# yet or that have been deleted so it can't just depend on all the
+# requirements/*.txt files that exist on disk $(wildcard requirements/*.txt).
+#
+# Instead we generate the list of requirements/*.txt files by getting all the
+# requirements/*.in files from disk ($(wildcard requirements/*.in)) and replace
+# the .in's with .txt's.
+.PHONY: requirements requirements/
+requirements requirements/: $(foreach file,$(wildcard requirements/*.in),$(basename $(file)).txt)
```
Comment (seanh) [inline Makefile:108]: This adds the `make requirements` command for (re-)compiling `requirements/*.txt`, same as `h-periodic` and all our other projects have.

File: requirements/coverage.in [hunk 2]
```diff
@@ -1 +1,3 @@
+pip-tools
+pip-sync-faster
```
Comment (seanh) [inline requirements/coverage.in:2]: `make requirements` and the new `tox.ini` stuff below require `pip-tools` and `pip-sync-faster` to be installed in all the tox environments so they're added to all the `requirements/*.in` files. (Except the `requirements/requirements.in` file that's used for the production Docker container: we don't need `pip-tools` and `pip-sync-faster` to be installed in production, they're just dev tools.)

File: requirements/coverage.txt [hunk 3]
```diff
@@ -1,8 +1,36 @@
 #
-# This file is autogenerated by pip-compile
+# This file is autogenerated by pip-compile with python 3.8
 # To update, run:
 #
-#    pip-compile requirements/coverage.in
+#    pip-compile --allow-unsafe requirements/coverage.in
```
Comment (seanh) [inline requirements/coverage.txt:5]: These changes to the comments at the tops of the `requirements/*.txt` files are due to them having been recompiled with `make requirements` which uses a slightly different `pip-compile` command than `hdev requirements` did.

File: requirements/coverage.txt [hunk 4]
```diff
@@ -1,8 +1,36 @@
...
 #
+build==0.8.0
+    # via pip-tools
+click==8.1.3
+    # via pip-tools
```
Comment (seanh) [inline requirements/coverage.txt:10]: The packages that're being added to the `requirements/*.txt` files are due to `pip-tools`, `pip-sync-faster` and their dependencies having been added. Note that the production `requirements.txt` file hasn't changed so this doesn't make any difference to QA or production

**Review by indigobravo — APPROVED** [review]
Thanks Sean. Approved.

**Pull request merged**